specs/HubProtocol.md

```diff
@@ -20,13 +20,17 @@ This document describes two encodings of the SignalR protocol: [JSON](http://www
 
 In the SignalR protocol, the following types of messages can be sent:
 
-* `Negotiation` Message - Sent by the client to negotiate the message format.
-* `Invocation` Message - Indicates a request to invoke a particular method (the Target) with provided Arguments on the remote endpoint.
-* `StreamInvocation` Message - Indicates a request to invoke a streaming method (the Target) with provided Arguments on the remote endpoint.
-* `StreamItem` Message - Indicates individual items of streamed response data from a previous Invocation message.
-* `Completion` Message - Indicates a previous Invocation or StreamInvocation has completed. Contains an error if the invocation concluded with an error or the result of a non-streaming method invocation. The result will be absent for `void` methods. In case of streaming invocations no further `StreamItem` messages will be received
-* `CancelInvocation` Message - Sent by the client to cancel a streaming invocation on the server.
-* `Ping` Message - Sent by either party to check if the connection is active.
+| Message Name          | Sender         | Description                                                                                                                    |
+| ------------------    | -------------- | ------------------------------------------------------------------------------------------------------------------------------ |
+| `NegotiationRequest`  | Client         | Sent by the client to negotiate the message format.                                                                            |
+| `NegotiationResponse` | Server         | Sent by the server as an acknowledgment of the previous `NegotiationRequest` message. Contains an error if negotiation failed. |
+| `Close`               | Callee, Caller | Sent by the server when a connection is closed. Contains an error if the connection was closed because of an error.            |
```

Comment:
We'll also need to add information to the encoding specifics further down the spec for the new message types.

Reply:
Close is Callee and Caller?

Reply:
Correct. It can be sent by either node. Since there's no invocation context, it would probably be more accurate to say "Client, Server". This is one of many reasons I'm no longer in love with "Caller/Callee" :)